lib/travis/build/git/clone.rb

```diff
@@ -45,12 +45,18 @@ def fetch_ref?
           end
 
           def checkout
-            sh.cmd "git checkout -qf #{data.pull_request ? 'FETCH_HEAD' : data.commit}", timing: false
+            sh.cmd "git checkout -qf #{checkout_ref}", timing: false
+          end
+
+          def checkout_ref
+            return 'FETCH_HEAD' if data.pull_request
+            return data.tag     if data.tag
+            data.commit
           end
 
           def clone_args
             args = "--depth=#{depth}"
-            args << " --branch=#{branch}" unless data.ref
+            args << " --branch=#{branch}" unless data.ref || data.tag
```

Comment:
@BanzaiMan using `--branch=` for tags will work for the majority of cases. I believe branches and tags with the same name is fairly rare. That said, if we want to protect against this then we will need to change how we clone and checkout tags.